### ISSUE TITLE
vcpkg is not integrated into VS2017

### ISSUE THREAD

**Issue by mmatrosov** [Microsoft/vcpkg#707]
I did all the steps as shown in `readme.md`. Everything works fine in VS2015, but not in VS2017. How do I integrate install into VS2017?

**Comment by ras0219-msft** [comment]
VS2017 integration should activate using the same command as VS2015 (`vcpkg integrate install`).

Could you provide more details about your specific situation? We do have a known issue where VS2015 projects loaded inside VS2017 will _not_ have Vcpkg available, so please confirm whether the integration works with a freshly created project. If this is the bug affecting you, our current workaround is to use the per-project integration to fix it (`vcpkg integrate project`) [1].

[1] https://github.com/Microsoft/vcpkg/blob/master/docs/EXAMPLES.md#example-1-2-d

**Comment by mmatrosov** [comment]
What details do you need? I created a new Visual C++ empty project. There is no `$(VcpkgRoot)` available.

**Comment by ras0219-msft** [comment]
Thanks.

To triple check that none of the most common issues apply:
- Run `vcpkg install zlib:x86-windows`
- Inside your VS 2017 empty project, add a main file that uses `#include "zlib.h"`
- Make sure you're building Debug x86.
- Try building.
- If it fails, go to Tools -> Options -> Projects and Solutions -> Build and Run and set the output verbosity to Normal
- Run a rebuild and paste the output window's contents here; I'm specifically looking for a section like:
```
1>Target VcpkgTripletSelection:
1>  Using triplet "x86-windows" from "D:\src\vcpkg\installed\x86-windows\"
```

**Comment by mmatrosov** [comment]
Hmm, it looks very strange, but seems like doing these steps fixed (almost) my problem. I can now build my solution in VS2017 that was failing to build before. Did `vcpkg install zlib:x86-windows` cure it somehow?.. See the log of this operation:

> e:\work\vcpkg>vcpkg install zlib:x86-windows > log
> 
> Download cmake
> No suitable version of cmake was found (requires 3.8.0 or higher). Download
> portable version?
> [Y] Yes  [N] No  [A] Always Allow Downloads  [?] Help (default is "Y"): y
> 
> E:\work\vcpkg\downloads\cmake-3.8.0-rc1-win32-x86.zip
> -- CURRENT_INSTALLED_DIR=E:/work/vcpkg/installed/x86-windows
> -- DOWNLOADS=E:/work/vcpkg/downloads
> -- CURRENT_PACKAGES_DIR=E:/work/vcpkg/packages/zlib_x86-windows
> -- CURRENT_BUILDTREES_DIR=E:/work/vcpkg/buildtrees/zlib
> -- CURRENT_PORT_DIR=e:/work/vcpkg/ports/zlib/.
> -- Downloading http://www.zlib.net/zlib-1.2.11.tar.gz...
> -- Downloading http://www.zlib.net/zlib-1.2.11.tar.gz... OK
> -- Testing integrity of downloaded file...
> -- Testing integrity of downloaded file... OK
> -- Extracting source E:/work/vcpkg/downloads/zlib1211.tar.gz
> -- Extracting done
> -- Configuring x86-windows-rel
> -- Configuring x86-windows-rel done
> -- Configuring x86-windows-dbg
> -- Configuring x86-windows-dbg done
> -- Package x86-windows-rel
> -- Package x86-windows-rel done
> -- Package x86-windows-dbg
> -- Package x86-windows-dbg done
> -- Installing: E:/work/vcpkg/packages/zlib_x86-windows/share/zlib/copyright
> -- Performing post-build validation
> -- Performing post-build validation done
> Package zlib:x86-windows is installed

Note also how this question about CMake appears regardless of whether I have CMake installed on my machine or not, I already faced it before.

Now the only problem I see that `$(VcpkgRoot)` is not being displayed as inherited value in "Additional Include Directories". While the build system is clearly accessing it. I can also see all vcpkg-related macros available when I click "Macros>>" in the window for editing "Additional Include Directories". 

Same for Additional Library Directories - I cannot see any inherited values. I also can see neither includes nor libs path in command line in the project properties. Thus, it is quite difficult to reason, what header files and lib files are actually used.

**Comment by mmatrosov** [comment]
Again, it seems to be fully working now. It even copied necessary .dll files to my output directory. Namely, `boost_timer-vc140-mt-1_62.dll`. I'm building under x64 platform.

**Comment by ras0219-msft** [comment]
Really glad to hear that everything is working for you!

We recently needed to upgrade CMake to the Release Candidate for 3.8.0 to support VS 2017; it's very likely you don't have that installed system-wide. If I had to guess, you probably currently have a version older than 3.7.2 installed, which is why you've experienced two prompts for CMake (we used to require 3.7.2, then we recently upgraded the requirement to 3.8.0).

Unfortunately, whether variables are displayed in various parts of the UI is determined by the specific location inside the `vcxproj` that they are set. We (for reasons I'll omit here) use different integration points in VS2015 and VS2017, which is why the UI reports different results.

We do want to fix that issue in the future (by moving around the integration point), however it will require a _lot_ of testing and it's quite likely that user projects will still encounter instability. It just hasn't hit the right combination of benefit vs cost vs risk vs other things we can improve :).
